docs/ECMAScriptModules.md

```diff
@@ -38,4 +38,51 @@ import.meta.jest.useFakeTimers();
 // jest === import.meta.jest => true
 ```
 
+Additionally, since ESM evaluates static `import` statements before looking at the code, hoisting on `jest.mock` calls that happens in CJS modules won't work in ESM. To `mock` modules in ESM, you need to use dynamic `import()` after `jest.mock` calls to load the mocked modules, same applies to modules which have to load the mocked modules.
+
+```js title="main.cjs"
+const { BrowserWindow, app } = require('electron');
+
+// etc.
+
+module.exports = { example };
+```
+
+```js title="main.test.cjs"
+import { createRequire } from 'node:module';
+import { jest } from '@jest/globals';
+
+const require = createRequire(import.meta.url);
+
+jest.mock('electron', () => ({
```

Comment:
note that this only works because you use `require` in "production" code - if that was using `import` (which I would think everybody reading these docs are doing) then it won't work. So I think we should have the unstable API first, then go on to say that an alternative is to use CJS wrappers and mock them

Reply:
You do have a point there. Lemme rearrange the order of the examples there and maybe change some of the wordings to make it better to read.